### PR TITLE
[CLOUDTRUST-3991] kafka audit event persistency

### PR DESCRIPTION
**Pull request by basbeu**

### CODE REVIEW

**Review by harture — APPROVED** [review]

File: pkg/account/component_test.go [hunk 1]
```diff
@@ -219,6 +218,8 @@ func TestUpdateAccount(t *testing.T) {
 		Locale:      &locale,
 	}
 
+	mocks.auditEventsReporterModule.EXPECT().ReportEvent(gomock.Any(), gomock.Any()).AnyTimes()
```
Comment (fperot74) [inline pkg/account/component_test.go:221]: We don't want to check which event is emitted anymore?

File: pkg/account/component_test.go [hunk 2]
```diff
@@ -523,10 +513,10 @@ func TestUpdateAccountRevokeAccreditation(t *testing.T) {
 				assert.Equal(t, revokedAccred, kcUserRep.GetAttribute(constants.AttrbAccreditations)[0])
 				return nil
 			}).Times(1)
-		mocks.eventDBModule.EXPECT().ReportEvent(ctx, "UPDATE_ACCOUNT", "self-service", gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).Times(1)
+		mocks.auditEventsReporterModule.EXPECT().ReportEvent(ctx, gomock.Any()).Times(1)
```
Comment (fperot74) [inline pkg/account/component_test.go:516]: Don't need to specify .Times(1) (we used to write it but we are removing them now)

File: pkg/kyc/component_test.go [hunk 3]
```diff
@@ -420,28 +419,14 @@ func TestValidateUser(t *testing.T) {
 		mocks.keycloakClient.EXPECT().UpdateUser(accessToken, targetRealm, userID, gomock.Any()).Return(nil)
 		mocks.usersDB.EXPECT().StoreOrUpdateUserDetails(ctx, targetRealm, gomock.Any()).Return(nil)
 		mocks.usersDB.EXPECT().CreateCheck(ctx, targetRealm, userID, gomock.Any()).Return(nil)
-		mocks.eventsDB.EXPECT().ReportEvent(gomock.Any(), "VALIDATE_USER", "back-office", gomock.Any())
+		mocks.eventsReporter.EXPECT().ReportEvent(gomock.Any(), gomock.Any())
```
Comment (fperot74) [inline pkg/kyc/component_test.go:422]: Checking the event type can be useful when you need to maintain this code. (you know which event is expected)

File: pkg/register/component.go [hunk 4]
```diff
@@ -278,13 +286,13 @@ func (c *component) createUser(ctx context.Context, accessToken string, realmNam
 	return *kcUser.ID, *kcUser.Username, nil
 }
 
-func (c *component) reportEvent(ctx context.Context, apiCall string, values ...string) {
+/*func (c *component) reportEvent(ctx context.Context, apiCall string, values ...string) {
```
Comment (fperot74) [inline pkg/register/component.go:289]: Not used anymore... can be removed

**Review by fperot74 — APPROVED** [review]

**Pull request merged**